cognee/infrastructure/databases/relational/sqlalchemy/SqlAlchemyAdapter.py

```diff
@@ -89,10 +89,22 @@ async def delete_data_by_id(self, table_name: str, data_id: UUID, schema_name: O
         """
         Delete data in given table based on id. Table must have an id Column.
         """
-        async with self.get_async_session() as session:
-            TableModel = await self.get_table(table_name, schema_name)
-            await session.execute(TableModel.delete().where(TableModel.c.id == data_id))
-            await session.commit()
+        if self.engine.dialect.name == "sqlite":
+            async with self.get_async_session() as session:
+                TableModel = await self.get_table(table_name, schema_name)
+
+                # Foreign key constraints are disabled by default in SQLite (for backwards compatibility),
+                # so must be enabled for each database connection/session separately.
+                await session.execute(text("PRAGMA foreign_keys = ON;"))
```

Comment:
Why do we do this here? I would expect this would happen during migration/startup?

Reply:
It must be set for each new connection/session. It doesn't work if it's only done during migration/startup, I've tried.

Though some people use event listeners to execute the PRAGMA command for each new session/connection for this reason.